mongoimport/mongoimport.go

```diff
@@ -466,18 +460,12 @@ func (imp *MongoImport) runInsertionWorker(readDocs chan bson.D) (err error) {
 	}
 	collection := session.Database(imp.ToolOptions.DB).Collection(imp.ToolOptions.Collection)
 
-	var inserter flushInserter
-	if imp.IngestOptions.Mode == modeInsert {
-		bufferedInserter := db.NewBufferedBulkInserter(collection, imp.IngestOptions.BulkBufferSize, !imp.IngestOptions.StopOnError)
-		bufferedInserter.SetBypassDocumentValidation(imp.IngestOptions.BypassDocumentValidation)
-		if !imp.IngestOptions.MaintainInsertionOrder {
-			bufferedInserter.Unordered()
-		}
-
-		inserter = bufferedInserter
-	} else {
-		inserter = imp.newUpserter(collection, imp.IngestOptions.BypassDocumentValidation)
+	inserter := db.NewBufferedBulkInserter(collection, imp.IngestOptions.BulkBufferSize, !imp.IngestOptions.StopOnError)
+	inserter.SetBypassDocumentValidation(imp.IngestOptions.BypassDocumentValidation)
+	if !imp.IngestOptions.MaintainInsertionOrder {
+		inserter.Unordered()
 	}
+	inserter.SetUpsert(true)
```

Comment:
In both the replace and update cases, we want to do upserts. It's ignored by the insert case because the insert command doesn't have an upsert option.